### ISSUE TITLE
KLqp accuracy issue?

### ISSUE THREAD

**Issue by lazypanda1** [blei-lab/edward#869]
Running the program below gives me incorrect result. I expected `qw` to be around `9` for this dataset. But the results are not close.

Output:
```
3.5601854
0.29567945
21.178354
```

Code:
```python

import edward as ed, tensorflow as tf, numpy as np
datax = np.array([28.74868970158939, 2.8703336089566545, 39.5682961248604, 80.15458231027912, 84.75586141530879])
datay = np.array([264.73822, 31.833002, 362.11465, 727.39124, 768.80273])
datax = datax.reshape((5, 1))
ed.set_seed(100)
X = tf.placeholder(tf.float32, [5, 1])
p168 = ed.models.Beta((tf.ones(1) * 23.226223140121917), (tf.ones(1) * 6.185129652837107))
p169 = ed.models.Gamma((tf.ones(1) * 62.07880607105276), (tf.ones(1) * 82.62035198114357))
p170 = ed.models.Exponential((tf.ones(1) * 18.13589757166789))
y = ed.models.Normal((ed.dot(X, p168) + p169), p170)
qw = ed.models.Normal(tf.Variable(tf.random_normal([1])), tf.Variable(tf.random_normal([1])))
qb = ed.models.Normal(tf.Variable(tf.random_normal([1])), tf.Variable(tf.random_normal([1])))
qe = ed.models.Normal(tf.Variable(tf.random_normal([1])), tf.Variable(tf.random_normal([1])))
inference = ed.KLqp({p168: qw, p169: qb, p170: qe}, data={X: datax, y: datay})
import sys

inference.run()
print(qw.eval())
print(qb.eval())
print(qe.eval())
```
Is there something wrong with the model here? Or is this an accuracy issue with KLqp? For similar model in pyro, I get the correct results.

**Comment by mortonjt** [comment]
Not sure what is going on here, I have noticed `KLqp` misbehave myself (possibly due to the large gradient updates from the black-box inference).  

There are a couple of mistakes I can see off the bat though.

```
qw = ed.models.Normal(tf.Variable(tf.random_normal([1])), tf.Variable(tf.random_normal([1])))
qb = ed.models.Normal(tf.Variable(tf.random_normal([1])), tf.Variable(tf.random_normal([1])))
qe = ed.models.Normal(tf.Variable(tf.random_normal([1])), tf.Variable(tf.random_normal([1])))
```

That should actually be
```
qw = ed.models.Normal(tf.Variable(tf.random_normal([1])), 
                                  tf.nn.softplus(tf.Variable(tf.random_normal([1]))))
qb = ed.models.Normal(tf.Variable(tf.random_normal([1])), 
                                  tf.nn.softplus(tf.Variable(tf.random_normal([1]))))
qe = ed.models.Normal(tf.Variable(tf.random_normal([1])), 
                                  tf.nn.softplus(tf.Variable(tf.random_normal([1]))))
```
You can't feed in negative variances into the normal distribution.

**Comment by lazypanda1** [comment]
Tried that... but I still get something like this 
```
[ 5.10073662]
[ 4.35015678]
[ 7.02902555]
```
